bundle/schema/schema.go

```diff
@@ -9,6 +9,14 @@ import (
 	"github.com/databricks/cli/libs/jsonschema"
 )
 
+// Fields tagged "readonly" should not be emitted in the schema as they are
+// computed at runtime, and should not be assigned a value by the bundle author.
+const readonlyTag = "readonly"
+
+// Annotation for internal bundle fields that should not be exposed to customers.
+// Fields can be tagged as "internal" to remove them from the generated schema.
+const internalTag = "internal"
```

Comment:
Aren't practically readonly and internal are the same fields? Both of them sounds like private fields which are only used by internals of CLI, what the difference between 2 of them?

Reply:
@shreyas-goenka does it include making sure that internal fields can't be used in interpolation? otherwise the behaviour is identical

Reply:
I think it's fine if we allow interpolation for internal fields. The problem this solves is to avoid customers seeing these fields, and then wondering what `internal` fields do. Yes, the behavior is identical, but the benefit we get is that the intention behind why the field is hidden is clear.

Reply:
As we implement customer unmarshaling/config loading, we can use the `readonly` tag as well to prevent users from setting it. +1 that that the different intent warrants a different tag.